frontend/src/scenes/insights/ActionFilter/RenameModal.tsx

```diff
@@ -1,65 +1,50 @@
 import { useActions, useValues } from 'kea'
 import { entityFilterLogic } from 'scenes/insights/ActionFilter/entityFilterLogic'
-import React, { useEffect, useRef, useState } from 'react'
+import React, { useEffect, useRef } from 'react'
 import { EntityFilter, InsightType, ViewType } from '~/types'
 import { Button, Input, Modal } from 'antd'
 import { getDisplayNameFromEntityFilter } from 'scenes/insights/utils'
+import { renameModalLogic } from 'scenes/insights/ActionFilter/renameModalLogic'
+import { InputFocusOptions } from 'antd/es/input/Input'
 
 interface Props {
-    visible: boolean
-    setModalOpen: (state: boolean) => void
+    typeKey: string
     view?: InsightType
 }
 
-export function FilterRenameModal({ visible, setModalOpen, view }: Props): JSX.Element {
-    const ref = useRef<Input | null>(null)
-    const { selectedFilter } = useValues(entityFilterLogic)
-    const { renameFilter, selectFilter } = useActions(entityFilterLogic)
-
-    const [name, setName] = useState(getDisplayNameFromEntityFilter(selectedFilter) ?? '')
+export function RenameModal({ typeKey, view }: Props): JSX.Element {
+    const { selectedFilter, modalVisible } = useValues(entityFilterLogic)
+    const { renameFilter, setModalVisible } = useActions(entityFilterLogic)
 
-    useEffect(() => {
-        setName(getDisplayNameFromEntityFilter(selectedFilter) ?? '')
-    }, [selectedFilter])
+    const logic = renameModalLogic({ typeKey, filter: selectedFilter })
+    const { name } = useValues(logic)
+    const { setName } = useActions(logic)
 
-    // Hacky setTimeout is needed - https://github.com/ant-design/ant-design/issues/8668#issuecomment-352955313
-    useEffect(() => {
-        const autoFocusTimeout = setTimeout(() => {
-            if (visible && ref.current) {
-                ref.current?.focus({
-                    cursor: 'all',
-                })
-            }
-        }, 0)
-        return () => clearTimeout(autoFocusTimeout)
-    }, [visible])
+    const ref = useRef<Input | null>(null)
+    useSelectAllText(ref, { cursor: 'all' }, [modalVisible])
 
     const onRename = (): void => {
         renameFilter({ ...selectedFilter, custom_name: name } as EntityFilter)
-        setModalOpen(false)
     }
```

Comment:
Could any of the logics also call this directly? Because as things stand now, this line is untested.

Reply:
good catch!

Reply:
I meant to remove the `renameFilter({ ...selectedFilter, custom_name: name } as EntityFilter)` line actually and move that into a logic, so it could also be tested.

All this data is accessible in a logic, so why should a React component know the inner details of how the filters object works?